src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -2136,3 +2144,23 @@ services:
     public: true
     tags:
       - { name: form.type }
+
+  form.type.improve.design.image_settings.image_generation_options_type:
+    class: 'PrestaShopBundle\Form\Admin\Improve\Design\ImageSettings\ImageGenerationOptionsType'
+    parent: 'form.type.translatable.aware'
+    public: true
+    arguments:
+      - '@=service("prestashop.adapter.shop.context").isAllShopContext()'
```

Comment:
can you inject the services instead? it avoids to have unneeded complexity in DI